execution_engine/src/engine_state/error.rs

```diff
@@ -162,6 +162,19 @@ impl From<stack::RuntimeStackOverflow> for Error {
     }
 }
 
+impl From<Error> for binary_port::ErrorCode {
+    fn from(err: Error) -> Self {
+        match err {
+            Error::RootNotFound(_) => binary_port::ErrorCode::RootNotFound,
+            Error::InvalidDeployItemVariant(_) => binary_port::ErrorCode::InvalidDeployItemVariant,
+            Error::WasmPreprocessing(_) => binary_port::ErrorCode::WasmPreprocessing,
+            Error::InvalidProtocolVersion(_) => binary_port::ErrorCode::UnsupportedProtocolVersion,
+            Error::Deploy => binary_port::ErrorCode::InvalidDeploy,
+            _ => binary_port::ErrorCode::InternalError,
```

Comment:
we've mapped only the necessary error information to cover the existing error codes used in the RPC, we could want to handle all the errors, but most of them are things the sidecar currently doesn't care about

Reply:
Yes, this feature was initially defined as "move RPC out of scope", hence the decision to forward only the information strictly required for a sidecar to function as the "old" RPC server.

Later the scope was creeping towards a fully-fledged binary port access for the node, which could justify the need to forward as details information as we can to allow the client to interpret it.

This might actully be a good point, but I'll suggest doing it as a separate PR/ticket.

Reply:
Fair enough.  I'll mark the comments as resolved regarding the error-reporting, assuming it will get handled in a later PR.